test/pg/postgres_test.exs

```diff
@@ -47,6 +54,28 @@ defmodule EctoEnum.PostgresTest do
     assert changes.status == :inactive
   end
 
+  test "casts atom to binary when enums are strings" do
+    %{errors: errors} = Ecto.Changeset.cast(%User{}, %{"role" => 3}, ~w(role)a)
+    error = {:role, {"is invalid", [type: EctoEnum.PostgresTest.RoleEnum, validation: :cast]}}
+    assert error in errors
+
+    %{changes: changes} = Ecto.Changeset.cast(%User{}, %{"role" => "manager"}, ~w(role)a)
+    assert changes.role == "manager"
+
+    %{changes: changes} = Ecto.Changeset.cast(%User{}, %{"role" => :admin}, ~w(role)a)
+    assert changes.role == "admin"
+  end
+
+  test "loads enum in type defined in defenum/3" do
+    status = :active
```

Comment:
I think it would be valuable to test that this happens even if we create using status as string and role as atom